core/src/main/java/org/apache/spark/TaskContext.java

```diff
@@ -94,6 +95,20 @@ static void unset() {
   @Deprecated
   public abstract void addOnCompleteCallback(final Function0<Unit> f);
 
+  /**
+   * Add a listener to be executed when kill a task. We add this
```

Comment:
Please edit these comments to more closely match the style of addTaskCompletionListener, e.g. to say that this is a Java-friendly interface and that the other method takes a Scala closure.